pkg/worker/worker.go

```diff
@@ -44,19 +44,25 @@ func init() {
 // 3. The done file contains a single string of the results to be sent to the master
 func GatherResults(waitfile string, url string, client *http.Client) error {
 	logrus.WithField("waitfile", waitfile).Info("Waiting for waitfile")
-	stop := sigHandler()
+	signals := sigHandler()
 	ticker := time.Tick(1 * time.Second)
+	stop := make(chan struct{}, 1)
 	for {
 		select {
 		case <-ticker:
 			if resultFile, err := ioutil.ReadFile(waitfile); err == nil {
 				logrus.WithField("resultFile", string(resultFile)).Info("Detected done file, transmitting result file")
 				return handleWaitFile(string(resultFile), url, client)
 			}
+		case <-signals:
+			// Run a goroutine here so we can keep checking the done file before cleaning up.
+			go func() {
+				time.Sleep(plugin.GracefulShutdownPeriod)
+				stop <- struct{}{}
```

Comment:
can't you just set a timer for the stop channel using wait.Until()? 

Looks at upstream examples.

Reply:
https://github.com/kubernetes/apimachinery/blob/master/pkg/util/wait/wait.go

Reply:
We could do that, but we still have to coordinate three cases here.

1) We got the done file
2) We got a SIGINT
3) We got a "ok shut everything down message"

If I were to wait.Until() with some callback that executes every period until the stop channel passed in is closed, I think it would be harder to read, but I'm more used to the for/select.

It's your call. I literally wrote the code here so if you prefer this style, I can refactor.

This might also be a good suggestion for https://github.com/heptio/sonobuoy/issues/372 since now we've parameterized the callback as `func()`.

Example:
```go
stop := make(chan struct{}{})
signals :=  sigHandler()
go func() {
    <-signals
    time.Sleep(plugin.GracefulShutdownPeriod)
    close(stop)
}
wait.Until(func(){
    if resultFile, err := ioutil.ReadFile(waitfile); err == nil {
         logrus.WithField("resultFile", string(resultFile)).Info("Detected done file, transmitting result file")
        return handleWaitFile(string(resultFile), url, client)
    }
}, 1 *time.Second, stop)
```

Reply:
I'm ok with the fix, please add a `//TODO evaluate wait.Until time permitting` and we can move onto other things.